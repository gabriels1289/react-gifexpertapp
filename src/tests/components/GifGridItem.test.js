import { GifGridItem } from "../../components/GifGridItem";
import { render } from '@testing-library/react';
import { shallow } from 'enzyme'; 

describe('Pruebas GifGridItem', () => {

  const title = 'Un título';
  const url = 'https://localhost/algo.jpg';

  let wrapper = shallow( <GifGridItem title={title} url={url} /> );

  beforeEach( () => {
    wrapper = shallow( <GifGridItem title={title} url={url} /> );
  });

  test('Debe de mostrar <GifGridItem /> correctamente (hacer match con snapshot)', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe de tener un párrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Debe de tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    // img.props().src
    // img.prop('src')
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe de tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  });

})