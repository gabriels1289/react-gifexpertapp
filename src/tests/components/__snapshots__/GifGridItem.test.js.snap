// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGridItem Debe de mostrar <GifGridItem /> correctamente (hacer match con snapshot) 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un título
     
  </p>
</div>
`;
